### PR TITLE
Uml 1192 older lpa api

### PR DESCRIPTION
**Pull request by cooperaj**
# Purpose

Add and endpoint to the api that allows us to request the sending of access code letters via the sirius api

Fixes UML-1192
Fixes UML-1193
Fixes UML-1194

## Approach

1) Rewrite large chunks of the PACT testing library
2) Redo a significant number of integration test steps to use the new pact library
3) Add the letter request to the lpa layer
4) Add the handler

## Learning

The PACT mock stuff is tricky to wrangle. We should look into not using behat tests for Integration.

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] ~I have added welsh translation tags and updated translation files~
* [ ] ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* [ ] ~The product team have tested these changes~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,64 +1,112 @@
-THIS_FILE := $(lastword $(MAKEFILE_LIST))
-.PHONY: everything rebuild down destroy ps logs up_dependencies up_service up_seeding
+COMPOSE = docker-compose -f docker-compose.yml -f docker-compose.dependencies.yml
+OVERRIDE := $(shell find . -name "docker-compose.override.yml")
+ifdef OVERRIDE
```
Comment (cooperaj) [inline Makefile:3]: Add an override file to the docker-compose command if it exists

File: Makefile [hunk 2]
```diff
@@ -1,64 +1,112 @@
...
+.PHONY: up
 
 exec:
-	docker-compose -f docker-compose.yml -f docker-compose.dependencies.yml exec $(c)
+	$(COMPOSE) exec $(c)
+.PHONY: exec
 
-up_all: | up_dependencies up_service
+# Starts the application and seeds initial data.
+up_all: | up_dependencies up_services seed
```
Comment (cooperaj) [inline Makefile:16]: Needed a way to start *everything* just like we currently do with the PHPStorm command

File: Makefile [hunk 3]
```diff
@@ -1,64 +1,112 @@
...
 build:
-	docker-compose -f docker-compose.yml -f docker-compose.dependencies.yml build
+	$(COMPOSE) build
+.PHONY: build
+
+build_all: build
+ifeq (, $(shell which go))
+	$(error "No golang in PATH, consider doing brew install go")
+endif
+	@echo "Installing go dependencies..."
```
Comment (cooperaj) [inline Makefile:27]: Integrate the getting started steps of the other repo here. Saves a bit of effort

File: service-api/app/src/App/src/Service/ActorCodes/Validation/CodesApiValidationStrategy.php [hunk 4]
```diff
@@ -38,7 +38,7 @@ public function validateCode(string $code, string $uid, string $dob): string
         try {
             $actorCode = $this->actorCodesApi->validateCode($code, $uid, $dob);
 
-            $actorUid = $actorCode->getData()['actor'] ?? null;
+            $actorUid = !empty($actorCode->getData()['actor']) ? $actorCode->getData()['actor'] : null;
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/ActorCodes/Validation/CodesApiValidationStrategy.php:41]: This is actually a bug that was being missed in our tests. The response for a validate code that fails is an empty actor value in an array. The null coalesce didn't work for this.

File: service-api/app/test/fixtures/example_lpa.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "receiptDate": "2018-06-30",
   "rejectedDate": "2018-06-30",
   "registrationDate": "2018-06-30",
-  "status": "Cancelled",
+  "status": "Registered",
```
Comment (cooperaj) [inline service-api/app/test/fixtures/example_lpa.json:9]: Our fixture data should really contain the case/s that the majority of our tests need and only make changes in the minority of cases. Anywhere in our contexts a 'Cancelled' status was being used it was already being set to that anyway.

**Comment by cooperaj** [comment]
Looks like I've somehow managed to omit Acceptance tests. Working on them.

File: service-api/app/features/context/Integration/LpaContext.php [hunk 6]
```diff
@@ -963,24 +958,12 @@ public function theLPAIsSuccessfullyAdded()
...
-        // this is now called twice
-        $this->apiFixtures->get('/v1/use-an-lpa/lpas/' . $this->lpaUid)
-            ->respondWith(
-                new Response(
-                    StatusCodeInterface::STATUS_OK,
-                    [],
-                    json_encode($this->lpa)
-                )
-            );
+//        $this->pactGetInteraction(
```
Comment (cooperaj) [inline service-api/app/features/context/Integration/LpaContext.php:961]: Why commented out?

**Review by GemTay — DISMISSED** [review]
Looking good 👍

**Review by andrewpearce-digital — APPROVED** [review]

**Review by SeemaMenon — APPROVED** [review]

**Pull request merged**